orderer/common/multichannel/registrar.go

```diff
@@ -537,7 +538,10 @@ func (r *Registrar) CreateChain(chainName string) {
 	if chain != nil {
 		logger.Infof("A chain of type %T for channel %s already exists. "+
 			"Halting it.", chain.Chain, chainName)
+		r.lock.Lock()
```

Comment:
Not strictly related to this PR, but I had a look and noticed that there are several accesses to `chains` that are also not gated by a lock. Could you perhaps double-check?

Reply:
There are two methods that use the locking mechanism to store `r.chain`s. Seems all new/refactored methods started to lock the chains map. I can go over whole register and add locks. 
@yacovm wdut ?

Reply:
Let's not do this in this PR.

Reply:
I can create another issue and do that. ok?

Reply:
The fact that we don't lock in every access doesn't mean we should. We don't need to lock in the main goroutine for memory access that chronologically precedes memory access by other goroutines, since all goroutines are children of the main goroutine, hence there is a happens-before memory relation.

I don't believe you will find a problem, but, I could be wrong, so if you find a missing lock, please do create an issue and fix it.

Reply:
Ok, marking it as resolved for now.

Reply:
Fine by me not to do it in this PR. I just find code that makes seemingly random use of locking for maps (with no explanation) a little hard to maintain. Ideally we'd always use locks (so it's easy to find issues) or carefully document things in the comments. We don't do either..